source/common/stats/stats_impl.cc

```diff
@@ -71,22 +94,11 @@ TagExtractorPtr TagExtractorImpl::createTagExtractor(const std::string& name,
     throw EnvoyException("tag_name cannot be empty");
   }
 
-  if (!regex.empty()) {
-    return TagExtractorPtr{new TagExtractorImpl(name, regex)};
-  } else {
-    // Look up the default for that name.
-    const auto& name_regex_pairs = Config::TagNames::get().name_regex_pairs_;
-    auto it = std::find_if(name_regex_pairs.begin(), name_regex_pairs.end(),
-                           [&name](const std::pair<std::string, std::string>& name_regex_pair) {
-                             return name == name_regex_pair.first;
-                           });
-    if (it != name_regex_pairs.end()) {
-      return TagExtractorPtr{new TagExtractorImpl(name, it->second)};
-    } else {
-      throw EnvoyException(fmt::format(
-          "No regex specified for tag specifier and no default regex for name: '{}'", name));
-    }
+  if (regex.empty()) {
+    throw EnvoyException(fmt::format(
+        "No regex specified for tag specifier and no default regex for name: '{}'", name));
```

Comment:
I know you just moved this, but as long as you're in here mind adding a unit test?

Reply:
done & done